GAD2.py

```diff
@@ -0,0 +1,209 @@
+import os
+import re
+from typing import TextIO, Optional, Union
+from abc import ABC, abstractmethod
+from Bio import SeqIO, SeqUtils
+
+class BiologicalSequence(ABC):
+    @abstractmethod
+    def __len__():
+        pass
+
+    def __getitem__():
+        pass
+
+    def __str__():
+        pass
+
+    def check_sequence():
+        pass
+
+class NucleicAcidSequence(BiologicalSequence):
+    """Proccising nucleic acid sequences.
+    This class is parental for DNASequence and RNASequence classes.
+    """
+    def __init__(self, seq) -> None:
+        self.seq = seq
+
+    def check_sequence(self):
+        """
+        Checkes the correctnessis of input string (DNA or RNA).
+        """
+        for nucleotide in self.seq:
+            if not nucleotide in type(self).complement_rule:
+                raise ValueError (f'Incorrect nucleotide in {self.seq}')
+        return True
+
+    def complement(self):
+        """
+        Create complement sequences of RNA or DNA acoording to the complementary base pairing rule. 
+        """
+        if self.check_sequence():
+            complement_sequence = ''
+            for nucleotide in self.seq:
+                if nucleotide in type(self).complement_rule:
+                    complement_sequence += type(self).complement_rule[nucleotide]
+            return type(self)(complement_sequence)
+
+    def gc_content(self):
+        return (sum(1 for _ in re.finditer(r'[GCgc]', self.seq)))/self.__len__()
```

Comment:
Оригинально! Ещё одно свидетельство того, как многогранен Пайтон.
Можно ещё такой вариант:

`(self.seq.upper().count('G') + self.seq.upper().count('C')) / len(self.seq)`
```suggestion
        return (sum(1 for _ in re.finditer(r'[GCgc]', self.seq))) / self.__len__()
```